providers/softlayer/src/main/java/org/jclouds/softlayer/compute/strategy/SoftLayerComputeServiceAdapter.java

```diff
@@ -447,6 +457,27 @@ public boolean apply(OperatingSystem input) {
               .first();
    }
 
+   private String getNotesFrom(SoftLayerTemplateOptions templateOptions) {
+      if (templateOptions.getNotes().isPresent()) {
+         return templateOptions.getNotes().get();
+      } else {
+         Map<String, String> meta = templateOptions.getUserMetadata();
+         if (meta != null && !meta.isEmpty()) {
+            if (meta.containsKey(USER_META_NOTES)) {
+               return meta.get(USER_META_NOTES);
+            } else {
+               String notes = "User Metadata\n===========\n\n" + Joiner.on("\n").withKeyValueSeparator(": ").join(meta);
+               if (notes.length() > SoftLayerTemplateOptions.NOTES_MAX_LENGTH) {
+                  String suffix = "...\n<truncated>";
+                  notes = notes.substring(0, SoftLayerTemplateOptions.NOTES_MAX_LENGTH - suffix.length()) + suffix;
+               }
+               return notes;
+            }
+         }
+      }
+      return null;
+   }
```

Comment:
This looks *very* confusing. Are there three different ways to set the notes? There should be only one understandable way to set them.
Also, what is the purpose of serialising the user metadata as notes? I mean, why the user metadata and not some other property?